src/extensions/video-plane-display-dmabuf.cpp

```diff
@@ -0,0 +1,346 @@
+
+                g_cond_signal(&threadSpawn.cond);
+                g_mutex_unlock(&threadSpawn.mutex);
+                return FALSE;
+            }, &threadSpawn, nullptr);
+        g_source_attach(source, context);
+        g_source_unref(source);
+    }
+
+    g_main_loop_run(loop);
+
+    g_main_context_pop_thread_default(context);
+    return nullptr;
+}
+
+GSourceFuncs DmaBufThread::PollingSource::s_sourceFuncs = {
+    // prepare
+    [](GSource* base, gint* timeout) -> gboolean
+    {
+        auto& source = *reinterpret_cast<PollingSource*>(base);
+
+        *timeout = -1;
+
+        if (source.isReading)
+            return FALSE;
+
+        // If there are pending dispatches on this queue we return TRUE to proceed to dispatching ASAP.
+        if (wl_display_prepare_read_queue(source.display, source.queue) != 0)
+            return TRUE;
+
+        source.isReading = true;
+
+        // We take up this opportunity to flush the display object, but
+        // otherwise we're not able to determine whether there are any
+        // pending dispatches (that would allow us skip the polling)
+        // without scheduling a read on the wl_display object.
+        wl_display_flush(source.display);
+        return FALSE;
+    },
+    // check
+    [](GSource* base) -> gboolean
+    {
+        auto& source = *reinterpret_cast<PollingSource*>(base);
+
+        // Only perform the read if input was made available during polling.
+        // Error during read is noted and will be handled in the following
+        // dispatch callback. If no input is available, the read is canceled.
+        // revents value is zeroed out in any case.
+        if (source.isReading) {
+            source.isReading = false;
+
+            if (source.pfd.revents & G_IO_IN) {
+                if (wl_display_read_events(source.display) == 0)
+                    return TRUE;
+            } else
+                wl_display_cancel_read(source.display);
+        }
+
+        return source.pfd.revents;
+    },
+    // dispatch
+    [](GSource* base, GSourceFunc, gpointer) -> gboolean
+    {
+        auto& source = *reinterpret_cast<PollingSource*>(base);
+
+        // Remove the source if any error was registered.
+        if (source.pfd.revents & (G_IO_ERR | G_IO_HUP))
+            return FALSE;
+
+        // Dispatch any pending events. The source is removed in case of
+        // an error occurring during this step.
+        if (wl_display_dispatch_queue_pending(source.display, source.queue) < 0)
+            return FALSE;
+
+        source.pfd.revents = 0;
+        return TRUE;
+    },
+    // finalize
+    [](GSource *base)
+    {
+        auto& source = *reinterpret_cast<PollingSource*>(base);
+
+        if (source.isReading) {
+            wl_display_cancel_read(source.display);
+            source.isReading = false;
+        }
+    },
+    nullptr, // closure_callback
+    nullptr, // closure_marshall
+};
```

Comment:
This si duplicated from TargetSource::s_sourceFuncs, could we share that somehow?

Reply:
Seems so :)